histogram_density.py

```diff
@@ -0,0 +1,103 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import cm
+
+from qibo import *
+
+
+def plot_density_hist(circuit):
```

Comment:
I believe this will be more flexible if it takes a state (as array) or a `CircuitResult` object. This would delegate the circuit execution to the user, instead of the plotting method:
```py
result = circuit.execute()
plot_density_hist(result)
```
which is useful in cases where the execution is customized, for example if one wants to pass initial state, nshots, etc.

Moreover, the state to be visualized may come from someting that is not a circuit, such variational models, etc..